components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -1472,6 +1477,47 @@ private void updateLocalAndOutboundAuthenticationConfiguration(int applicationId
         }
     }
 
+    private void updateAppRoleMappingConfiguration(int applicationId,
+                                                   AppRoleMappingConfig[] applicationRoleMappingConfigs,
+                                                   Connection connection)
+            throws SQLException {
+
+        int tenantID = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+
+        deleteAppRoleMappingConfiguration(applicationId, connection);
+
+        try (PreparedStatement updateAppRoleMappingPrepStmt = connection
+                .prepareStatement(ApplicationMgtDBQueries.UPDATE_SP_IDP_ATTR)) {
+            if (applicationRoleMappingConfigs != null) {
+                for (AppRoleMappingConfig applicationRoleMappingConfig : applicationRoleMappingConfigs) {
+
+                    // get idp id using idp name
+                    int idpId = getIdPId(connection, tenantID, applicationRoleMappingConfig.getIdPName());
+
+                    updateAppRoleMappingPrepStmt.setInt(1, applicationId);
+
```

Comment:
We can remove this extra line

Reply:
Addressed in 3da9288685f309f6395405363622af44c39f34c2